sbol_utilities/sbol3_sbol2_conversion.py

```
@@ -142,38 +145,58 @@ def visit_combinatorial_derivation(self, a: sbol3.CombinatorialDerivation):
-        cp2 = sbol2.ComponentDefinition(cp3.identity, types2, version=self._sbol2_version(cp3))
-        self.doc2.addComponentDefinition(cp2)
+        comp_def2 = sbol2.ComponentDefinition(comp3.identity, types2, version=self._sbol2_version(comp3))
+        self.doc2.addComponentDefinition(comp_def2)
         # Convert the Component properties not covered by the constructor
-        cp2.roles = cp3.roles
-        cp2.sequences = cp3.sequences
-        if cp3.features:
-            raise NotImplementedError('Conversion of Component features from SBOL3 to SBOL2 not yet implemented')
-        if cp3.interactions:
-            raise NotImplementedError('Conversion of Component interactions from SBOL3 to SBOL2 not yet implemented')
-        if cp3.constraints:
-            raise NotImplementedError('Conversion of Component constraints from SBOL3 to SBOL2 not yet implemented')
-        if cp3.interface:
+        comp_def2.roles = comp3.roles
+        comp_def2.sequences = comp3.sequences
+        if comp3.features:
+            for feature in comp3.features:
+                if type(feature) == sbol3.subcomponent.SubComponent:
+                    self.visit_sub_component(feature, comp_def2)
+                elif type(feature) == sbol3.compref.ComponentReference:
+                    try:
+                        self.visit_component_reference(feature)
+                    except NotImplementedError as e:
+                        # highlights the error message in red.
+                        print(f"\033[91m{e}\033[0m")
+                else:
+                    raise NotImplementedError(
+                        'Conversion of Component features from SBOL3 to SBOL2 not yet implemented')
+        if comp3.interactions:
+            for interaction in comp3.interactions:
+                try:
+                    self.visit_interaction(interaction)
+                except NotImplementedError as e:
+                    print(f"\033[91m{e}\033[0m")
```

Same as above - do not swallow errors here, or below